src/components/Form/TextInput.tsx

```diff
@@ -67,21 +67,65 @@ export class TextInput extends Component<TextInputProps> {
       ...passProps
     } = this.props;
 
-    const ifScreenreader = labelMode === 'screenreader';
-    if (ifScreenreader && !!placeholder) {
-      logger.warn(
-        'Placeholder not used as label is hidden and using label instead',
+    const hideLabel = labelMode === 'hidden';
+
+    return (
+      <HtmlLabel
+        {...labelProps}
+        className={classnames(labelBaseClassName, className, {
+          [disabledClassName]: !!passProps.disabled,
+        })}
+      >
+        {hideLabel ? (
+          <VisuallyHidden>{labelText}</VisuallyHidden>
+        ) : (
+          <Paragraph {...labelTextProps}>{labelText}</Paragraph>
+        )}
+        <HtmlDiv {...inputContainerProps}>
+          <HtmlInput
+            {...passProps}
+            className={classnames(inputBaseClassName, inputClassName)}
+            type="text"
+            placeholder={hideLabel ? labelText : !!placeholder ? labelText : ''}
+          />
+          {children}
+        </HtmlDiv>
+      </HtmlLabel>
+    );
+  }
+}
+
+export class TextInput extends BaseTextInput {
```

Comment:
This is written in this manner to allow the different logic for showing/disabling placeholder. If there's a cleaner way to utilize the BaseTextInput and overwrite the placeholder logic, let me know.

Reply:
So should this be using the `BaseTextInput` component passing props and only changing those props that are needed to be different?

Reply:
It was implemented like that at one point, but that would require figuring out a way to get the different placeholder visibility logic to the actual html component. Didn't get there yet. I'm open for suggestions.